test-harness/terratest-extensions/modules/azure/network.go

```diff
@@ -0,0 +1,48 @@
+package azure
+
+import (
+	"context"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
+	"testing"
+)
+
+func getVirtualNetworkClient(subscriptionID string) (*network.VirtualNetworksClient, error) {
+	authorizer, err := getServicePrincipalAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+
+	client := network.NewVirtualNetworksClient(subscriptionID)
+	client.Authorizer = authorizer
+	return &client, err
+}
+
+// GetSubnetsInVnetE - Return the subnets that exist wihin a given VNET
+func GetSubnetsInVnetE(subscriptionID string, resourceGroupName string, vnetName string) ([]string, error) {
+
+	client, err := getVirtualNetworkClient(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	vnet, err := client.Get(context.Background(), resourceGroupName, vnetName, "")
```

Comment:
I feel like we should pass in the context so that we can reuse the context throughout the tests. I also believe we could drop the bearer token in context.value and reuse it if needed. (I don't think we'll need to worry much about token timeout)

Reply:
I'd love to understand this more :). I'll be the first to admid that I went with "first thing to work" on this portion of the PR and would love to learn how to improve it.

I'll ramp up on what these contexts are before picking your brain.